### ISSUE TITLE
Treat type casts as used expressions

### ISSUE THREAD

**Issue by captbaritone** [eslint/eslint#12605]
Code like this is useful for forcing Flow to treat a switch statement as exhaustive:

```
function(foo: 'FOO' | 'BAR') {
  switch(foo) {
    case 'FOO':
      return true;
    case 'BAR';
      return false;
    default:
      // Currently this will trigger `no-unused-expressions`
      (foo: empty);
  }
}
```

But currently the type cast will trigger `no-unused-expressions` I'd like it not to.

**What rule do you want to change?**

no-unused-expressions

**Does this change cause the rule to produce more or fewer warnings?**

Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

New default behavior.

**Please provide some example code that this change will affect:**

```js

// This would not error.
(foo: empty);
```

**What does the rule currently do for this code?**

Currently this causes an error because the value is not used at runtime.

**What will the rule do after it's changed?**

The rule will not error on type casts.

**Are you willing to submit a pull request to implement this change?**

I am likely not the best person to make the actual change, but I could look into it if nobody on the core team is available.

**Comment by ilyavolodin** [comment]
ESLint doesn't support framework/library specific rules. So exceptions like that belong to plugins, not core rules. eslint-plugin-flowtype already has a rule that does exactly what you are looking for: https://github.com/gajus/eslint-plugin-flowtype#eslint-plugin-flowtype-rules-no-unused-expressions

**Comment by captbaritone** [comment]
Oh excellent. I didn’t know about that rule in the flowtype plugin. I’ll close this, since I think that solves my problem.

Thanks for all your hard work, it’s an amazing project.
